entrypoint.py

```diff
@@ -0,0 +1,127 @@
+
+import json
+import os
+
+import git
+import requests
+import semver
+
+
+def setup_git(merge_commit_sha):
+    """Initializes the git repo and checks out the commit ID created from the pull request.
+    Assumes the current directory is the git repo to be loaded.
+
+    Args:
+        merge_commit_sha (str): hex sha string of the merge commit
+
+    Returns:
+        repo (object): git repo object
+    """
+    repo = git.Repo(os.getcwd())
+    repo.git.checkout(merge_commit_sha)
+    return repo
+
+
+def semver_bump(repo):
+    """Loads the most recently created tag from the git repo and parses it into a semver object.
+    It is then incremented by a keyword in the commit message. Parsing exceptions are handled in parent function.
+
+    Args:
+        repo (object): git repo object
+
+    Returns:
+        new_tag (str): string of the new tag post incrementing semver section
+    """
+    current_tag = str(sorted(repo.tags, key=lambda t: t.commit.committed_datetime)[-1])
+    if current_tag.startswith('v'):
+        current_tag = current_tag[1:]
+
+    curr_ver = semver.VersionInfo.parse(current_tag)
+    commit_msg = repo.head.commit.message
+
+    if '#major' in commit_msg:
+        new_ver = curr_ver.bump_major()
+    elif '#minor' in commit_msg:
+        new_ver = curr_ver.bump_minor()
+    else:
+        new_ver = curr_ver.bump_patch()
+
+    if current_tag.startswith('v'):
+        return f'v{str(new_ver)}'
+    else:
+        return str(new_ver)
+
+
+def create_and_push_tag(repo, merge_commit_sha, new_tag):
+    """Creates a new tag from arguments in the git repo and pushes it to github.
+    Authentication is handled by environment variables passed in from github actions.
+
+    Args:
+        repo (object): git repo object
+        merge_commit_sha (str): hex sha string of merge commit used to attach tag
+        new_tag (str): new tag string to be committed and pushed to remote
+
+    Returns:
+        None
+    """
+    repo.create_tag(new_tag, ref=merge_commit_sha)
+    origin_url = f"https://{os.getenv('GITHUB_ACTOR')}:{os.getenv('GITHUB_TOKEN')}@github.com/{os.getenv('GITHUB_REPOSITORY')}.git"
+    gh_origin = repo.create_remote('github',  origin_url)
+    gh_origin.push(new_tag)
+
+
+def comment_on_pr(pr_number, comment_body):
+    """Comments on github PR with a given message.
+    Authentication is handled by environment variables passed in from github actions.
+
+    Args:
+        pr_number (int): number of the pull request. determines where to post message
+        comment_body (str): message to be posted to pull request by github actions bot
+
+    Returns:
+        None
+    """
+    url = f"https://api.github.com/repos/{os.getenv('GITHUB_REPOSITORY')}/issues/{pr_number}/comments"
+    headers = {
+        'Authorization': f"Bearer {os.getenv('GITHUB_TOKEN')}",
+        'Accept': 'application/vnd.github.v3+json'
+    }
+    body = {'body': comment_body}
+    requests.post(url, headers=headers, json=body)
```

Comment:
sucks that the `git` library doesn't support this.

Reply:
so, we could import the [PyGithub](https://pypi.org/project/PyGithub/) module which would likely be ably to do this but given that its a single call and that is a decently heavy module, I dont think its worth it.

Reply:
I lied, I wrote out what I think it would be, and I like it more.